### DISCUSSION TITLE
ArgoCD When connecting to Private Repo (via HTTPS), we are getting "repository not found"

### DISCUSSION THREAD

**Discussion by Josephred999** [argoproj/argo-cd#15623] [Q&A]
When Connecting from ArgoCD to HTTPS GitHub Private Repo, we are getting the below error:

`time="2023-09-21T13:43:56-04:00" level=fatal msg="rpc error: code = Unknown desc = error testing repository connectivity: repository not found"`

When I change the repo visibility from Private to Public, the connection is working fine. But the compliance standards are we need all of our repo's to be Private.

I had the root cert added for the authentication purposes, but ArgoCD should be able to connect to Private Repo's right?

Any info here will help my cause.

**Comment by FernandoMiguel** [comment]
when you mean private, you mean just a repo in github, or is there a network thing?
as long as the creds you are using have access to the repo, it should work.
what errors do you get when you try locally on your computer?

**Comment by patrickpnq2** [comment]
If this is still active. I had the same behaviour. 

> when you mean private, you mean just a repo in github, or is there a network thing?

Its the github.com setting for each repository. In this context, it means that the repository can only be accessed by someone within the github organisation that has been granted explicit access. 
Even after I granted my Github App access to the repository, I was receiving the error mentioned above.

`level=fatal msg="rpc error: code = Unknown desc = error testing repository connectivity: repository not found"`

The problem occurs only the first time to try to connect to any repository configured.

After you get the error when testing the connection in argocd, navigate to the github.com repositories setttings.
There is a button asking you to accept that your Github App is trying to access your repository.

Once this is validated, retrying the argocd connection works. Not only for this repo, but all other repositories that your Github App is permitted to access.

My guess is just that the http response code and/or error message sent by Github is either misleading, or the message is poorly handled in argocd's error stack.

It's a bit of an edge case.

**Reply by abourget** [reply]
I'm looking for a button, but can't find it. Do you have a more precise location? GitHub has accumulated quite a bit of settings pages over the years :)

**Reply by patrickpnq2** [reply]
I did not take a screenshot - and I haven't seen it since, since i think it only appears the first time.
But I would think one of the following locations is the answer. You might confirm which if you find it for others.

At the Repository Level under Settings -> GitHub Apps
At the Organisation Level under Settings -> GitHub Apps
At the Organisation Level under Settings -> Developer Settings -> GitHub Apps

**Comment by yohan-sentiva** [comment]
I faced the same issue and found that button here:

At the Repository Level under Settings -> GitHub Apps -> Configure and near 'Permissions a prompt was there asking me to review the permission change request.

**Reply by trishtechadmin** [reply]
Thanks for sharing the resolution steps for this issue, 
Seeing this issue  with my argocd microk8s installation with both pat and ssh key:

private repo: https://github.com/trishtechnologies0805-gmail-com/k8s-ansible.git
Verified pat has access to private repository using pat, I tried insecure flag in my values.yaml, this did not help

Unable to connect to repository: rpc error: code = Unknown desc = error testing repository connectivity: unable to ls-remote HEAD on repository: failed to list refs: Get "https://github.com/trishtechnologies0805-gmail-com/k8s-ansible.git/info/refs?service=git-upload-pack": remote error: tls: unrecognized name

with ssh key :
apiVersion: v1
data:
  sshPrivateKey: xxxxxxxxxxxx
  type: Z2l0
  url: Z2l0QGxxxxxx
  insecureIgnoreHostKey: xxxxxxxxx
kind: Secret
metadata:
  name: github-ssh-repo
  namespace: argocd
  labels:
    argocd.argoproj.io/secret-type: repository
---
# =============================================================================
# Known hosts for GitHub (prevents TOFU - Trust On First Use attacks)
# This is GitHub's SSH host key fingerprint
# =============================================================================
apiVersion: v1
kind: ConfigMap
metadata:
  name: argocd-ssh-known-hosts-cm
  namespace: argocd
  labels:
    app.kubernetes.io/name: argocd-ssh-known-hosts-cm
    app.kubernetes.io/part-of: argocd
data:
  ssh_known_hosts: |
    github.com ssh-ed25519 xxxxxxxxxx
    github.com ecdsa-sha2-nistp256 xxxxxxx
    github.com ssh-rsa xxxxxxxxxxxxxxxxxx
    
    #values,yaml
    `# ArgoCD Helm Values for MicroK8s with Nginx Proxy Manager
# Configured for base path /argocd to work behind reverse proxy
# Reference: https://github.com/argoproj/argo-cd/issues/2968

# Global settings
global:
  # Domain for ArgoCD - used in ingress and notifications
  domain: rhlab-argocd.example.net

# ArgoCD Configs
configs:
  # General ArgoCD settings
  cm:
    # Create admin user
    admin.enabled: "true"
    # Application status badge feature
    statusbadge.enabled: "true"
    # Enable anonymous access (optional, disable in production)
    users.anonymous.enabled: "false"
    # Enable exec feature for debugging
    exec.enabled: "true"
    # Server timeout settings
    timeout.reconciliation: 180s
    # Dex config (if using SSO later)
    dex.config: ""
    # Resource tracking method
    application.resourceTrackingMethod: annotation
    # URL configuration - CRITICAL for reverse proxy
    url: https://rhlab-argocd.htsvsys.net/argocd

  # ArgoCD parameters (creates argocd-cmd-params-cm ConfigMap)
  params:
    # CRITICAL: Run server in insecure mode (TLS terminated at proxy)
    server.insecure: true
    # CRITICAL: Base path for reverse proxy
    server.basehref: /argocd
    server.rootpath: /argocd
    # Disable TLS on gRPC (reverse proxy handles it)
    server.disable.auth: false
    # Static assets caching
    server.staticassets: /shared/app
    # Enable gzip compression
    server.enable.gzip: true
    # Disable TLS between ArgoCD internal components
    # Reference: https://argo-cd.readthedocs.io/en/stable/operator-manual/tls/
    reposerver.disable.tls: true
    server.repo.server.plaintext: true
    controller.repo.server.plaintext: true

  # RBAC Configuration
  rbac:
    # Policy for default permissions
    policy.default: role:readonly
    # Custom policies
    policy.csv: |
      p, role:admin, applications, *, */*, allow
      p, role:admin, clusters, *, *, allow
      p, role:admin, repositories, *, *, allow
      p, role:admin, logs, *, *, allow
      p, role:admin, exec, *, */*, allow
      g, admin, role:admin

  # Secret configuration
  secret:
    # Create the argocd-secret
    createSecret: true
    # Extra secrets (optional - add your tokens here)
    extra: {}

# Redis HA - disabled for simplicity (enable for HA setup)
redis-ha:
  enabled: false

# Disable built-in Redis - using external Redis server
redis:
  enabled: false

# External Redis Configuration - Using in-cluster Redis StatefulSet in rhlab-internal namespace
externalRedis:
  # Use local ExternalName service (argocd-redis-externalname.yaml) for cross-namespace access
  # The ExternalName service "redis" in argocd namespace points to redis-service.rhlab-internal
  ##k get svc redis -n argocd
  ## redis   ExternalName   <none>       redis-service.rhlab-internal.svc.cluster.local

  host: redis-service.rhlab-internal.svc.cluster.local.
  port: 6379
  password: ""  # Set via secret below
  existingSecret: argocd-redis-secret
  secretPasswordKey: redis-password

# ArgoCD Server
server:
  # Replicas
  replicas: 1

  # Resources
  resources:
    requests:
      cpu: 100m
      memory: 256Mi
    limits:
      cpu: 500m
      memory: 512Mi

  # Use default Kubernetes DNS (ClusterFirst) - no custom dnsPolicy
  # The ExternalName service handles cross-namespace resolution

  # Extra args for reverse proxy support (params ConfigMap handles most settings)
  extraArgs:
    - --basehref=/argocd
    - --rootpath=/argocd
    - --insecure

  # TLS certificate from cert-manager
  certificate:
    enabled: true
    secretName: argocd-server-tls
    domain: rhlab-argocd.htsvsys.net
    issuer:
      group: cert-manager.io
      kind: ClusterIssuer
      name: selfsigned-cluster-issuer
    additionalHosts:
      - argocd-server.argocd.svc
      - argocd-server.argocd.svc.cluster.local

  # Service configuration
  service:
    type: NodePort
    nodePortHttp: 30080
    nodePortHttps: 30443
    servicePortHttp: 80
    servicePortHttps: 443

  # Ingress - Traefik for web UI access
  ingress:
    enabled: true
    ingressClassName: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure,web
      traefik.ingress.kubernetes.io/router.tls: "true"
    hosts:
      - rhlab-argocd.htsvsystems.k8s.localnet
      - argocd-server.argocd.svc.cluster.local
      - argocd.k8s.local
    paths:
      - /argocd
    pathType: Prefix
    tls:
      - secretName: argocd-server-tls
        hosts:
          - argocd.k8s.local

  # Ingress for gRPC (CLI access)
  ingressGrpc:
    enabled: true
    ingressClassName: traefik
    annotations:
      traefik.ingress.kubernetes.io/router.entrypoints: websecure
      traefik.ingress.kubernetes.io/router.tls: "true"
      traefik.ingress.kubernetes.io/router.tls.certresolver: myresolver
    hosts:
      - rhlab-argocd.example.k8s.localnet
      - argocd-server.argocd.svc.cluster.local
      - argocd.k8s.local
    paths:
      - /grpc
      - /argocd/grpc
    pathType: Prefix
    tls:
      - secretName: argocd-server-tls
        hosts:
          - rhlab-argocd.examplek8s.localnet
          - argocd-server.argocd.svc.cluster.local

repoServer:
  # Use ClusterFirst DNS for internal service resolution (Redis, etc.)
  dnsPolicy: Default
  # Use hostNetwork to bypass network policies (if needed)
  replicas: 1
  # Environment variables to bypass network TLS inspection/proxy
  env:
    - name: GIT_SSL_NO_VERIFY
      value: "true"
    - name: GIT_TERMINAL_PROMPT
      value: "0"
    # Try to bypass transparent proxy for GitHub
    - name: NO_PROXY
      value: "github.com,*.github.com"
    - name: no_proxy
      value: "github.com,*.github.com"
  resources:
    requests:
      cpu: 200m
      memory: 512Mi
    limits:
      cpu: 1000m
      memory: 1Gi
  # Increase startup time for slow environments
  livenessProbe:
    initialDelaySeconds: 30
    periodSeconds: 10
    timeoutSeconds: 10
    failureThreshold: 6
  readinessProbe:
    initialDelaySeconds: 10
    periodSeconds: 10
    timeoutSeconds: 10
    failureThreshold: 3
  # TLS disabled via configs.params.reposerver.disable.tls

# Application Controller
controller:
  replicas: 1
  resources:
    requests:
      cpu: 250m
      memory: 512Mi
    limits:
      cpu: 1000m
      memory: 1Gi
  # Use default Kubernetes DNS (ClusterFirst)
  # Plaintext connection to repo-server via configs.params.controller.repo.server.plaintext

  # Metrics
  metrics:
    enabled: true
    serviceMonitor:
      enabled: false

# ApplicationSet Controller
applicationSet:
  enabled: true
  replicas: 1
  resources:
    requests:
      cpu: 100m
      memory: 128Mi
    limits:
      cpu: 200m
      memory: 256Mi

# Notifications Controller
notifications:
  enabled: true
  resources:
    requests:
      cpu: 100m
      memory: 128Mi
    limits:
      cpu: 200m
      memory: 256Mi
dex:
  enabled: false
haproxy:
  enabled: false
`

Can some one please help me here.
Thanks

**Comment by ethanbayliss** [comment]
1. make sure that your github app is installed in your account
2. make sure the github app is authorized into your repo

<img width="1134" height="358" alt="image" src="https://github.com/user-attachments/assets/43c7225f-4a81-43bd-b911-51d0440f2681" />
<img width="1279" height="1100" alt="image" src="https://github.com/user-attachments/assets/5faae51f-3036-42a1-b42a-11aa1603a6bf" />

**Comment by bkarankar** [comment]
just make sure you are using HTTPS based authentication, not using ssh based auth.
